PyPDF2/generic/_base.py

```diff
@@ -246,7 +246,7 @@ def __repr__(self) -> str:
             return str(self.quantize(decimal.Decimal(1)))
         else:
             # Standard formatting adds useless extraneous zeros.
-            o = f"{self:.5f}"
+            o = '{0:f}'.format(self)
             # Remove the zeros.
             while o and o[-1] == "0":
                 o = o[:-1]
```

Comment:
It looks like we could simplify this else branch down to just this one liner?
```
            return '{0:f}'.format(self).rstrip('0')
```

Reply:
Hmm, I on second thought there may be an edge case here, where we don't want to strip the last zero.

The while loop suffers the same problem as `rstrip` though, but shouldn't "0.000" become "0.0" instead of "0."?

Reply:
Oh, true.

something like:
```
o = '{0:f}'.format(self).rstrip('0')
if o[-1] == '.':
  o += "0"
return o
```

Reply:
Hm, though this also supposes that we'll always have a decimal number and not something like `Decimal('100')`

Reply:
Maybe (without testing):
```
o = '{0:f}'.format(self)
if o and '.' in o:
  o = o.rstrip('0')
  if o and o[-1] == '.':
    o += "0"
return o
```

Reply:
Ah, yep. Current implementation is bugged if the input is `Decimal("0")`!

Surprising that no tests caught that... I'll add a test that does.

Reply:
Dug into this one, and we're safe to simplify to `rstrip("0")`. The reason for that is because the first condition:

```python
        if self == self.to_integral():
            # If this is an integer, format it with no decimal place.
            return str(self.quantize(decimal.Decimal(1)))
```

I added the comment to help explain what the `to_integral` and `quantize` functions from `Decimal` are doing.

If the number is an integer, that covers where there's no decimal point or there's only zeros after the decimal e.g. `1000`, `1`, `1.0`, `1.0000`, etc

I've added a unit test `test_float_object_decimal_to_string` to test common and extreme cases.